systemverilog-plugin/UhdmAst.cc

```diff
@@ -834,9 +846,10 @@ static int simplify_struct(AST::AstNode *snode, int base_offset, AST::AstNode *p
             // embedded struct or union
             width = simplify_struct(node, base_offset + offset, parent_node);
             if (!node->multirange_dimensions.empty()) {
-                int number_of_structs = 1;
-                number_of_structs = node->multirange_dimensions.back();
-                width *= number_of_structs;
+                // reverse-iterate over multiranges, skip right_range
+                for (auto rit = node->multirange_dimensions.rbegin(); rit != node->multirange_dimensions.rend(); rit += 2) {
+                    width *= *rit;
+                }
```

Comment:
"skip right_range" → "skip offset"?
I think reverse iteration here is a bit too clever, doubly when emphasized in the comment.
```suggestion
                // Multiply widths of all dimensions.
                // `multirange_dimensions` stores (repeating) pairs of [offset, width].
                for (size_t i = 1; i < node->multirange_dimensions.size(); i += 2) {
                    width *= node->multirange_dimensions[i];
                }
```